release-22.1:  sql: fix telemetry panic for COPY

otan opened the pull request:
Backport 1/1 commits from #88322.

/cc @cockroachdb/release

---

Resolves #88229

Release note (bug fix): Fixed a bug where if telemetry is enabled, COPY can sometimes cause the server to crash.

Release justification: fixes a critical panic

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/88325)
<!-- Reviewable:end -->

cucaroach commented on pkg/sql/copy_in_test.go:
What's with this import?

otan commented on pkg/sql/copy_in_test.go:
My ide did that. I'll fix her up

otan committed:
sql: fix telemetry panic for COPY

Copy could previously crash if telemetry is turned on and it has to be
emitted to the sampling log. This unfortunately was not caught in
earlier tests as it was not sampled - in which the tests have now been
updated to force sampling.

Release note (bug fix): Fixed a bug where if telemetry is enabled, COPY
can sometimes cause the server to crash.
